proposals/partial-events-and-constructors.md

```diff
@@ -0,0 +1,293 @@
+     | 'virtual'
+     | 'sealed'
+     | 'override'
+     | 'abstract'
+     | 'extern'
++    | 'partial'
+     | unsafe_modifier
+     ;
+```
+
+Instance constructor declaration syntax ([§15.11.1][ctor-syntax]) is extended to allow the `partial` modifier:
+
+```diff
+ constructor_declaration
+     : attributes? constructor_modifier* constructor_declarator constructor_body
+     ;
+ 
+ constructor_modifier
+     : 'public'
+     | 'protected'
+     | 'internal'
+     | 'private'
+     | 'extern'
++    | 'partial'
+     | unsafe_modifier
+     ;
+```
+
+Note that the `partial` modifier can be specified anywhere among the modifiers unlike today in methods and types where it must be the last one.
+There is [a proposal][partial-ordering] to relax ordering of modifiers w.r.t. `partial` in method and type declarations as well.
+
+An event declaration with the `partial` modifier is said to be a *partial event declaration*
+and it is associated with one or more *partial events* with the specified name
+(note that one event declaration without accessors can define multiple events).
+
+A constructor declaration with the `partial` modifier is said to be a *partial constructor declaration*
+and it is associated with a *partial constructor* with the specified signature.
+
+A partial event declaration is said to be an *implementing declaration*
+when it specifies the `event_accessor_declarations` or it has the `extern` modifier.
+Otherwise, it is a *defining declaration*.
+
+A partial constructor declaration is said to be a *defining declaration*
+when it has the semicolon body and it lacks the `extern` modifier.
+Otherwise, it is an *implementing declaration*.
+
+```cs
+partial class C
+{
+    // defining declarations
+    partial C();
+    partial C(int x);
+    partial event Action E, F;
+
+    // implementing declarations
+    partial C() { }
+    partial C(int x) { }
+    partial event Action E { add { } remove { } }
+    partial event Action F { add { } remove { } }
+}
+```
+
+Only the defining declaration of a partial member participates in lookup
+and is considered at use sites and for emitting the metadata.
+(Except for documentation comments as detailed [below](#documentation-comments).)
+The implementing declaration signature is used when nullable-analyzing the associated bodies.
+
+A partial member:
+- May only be declared as a member of a partial type.
+- Must have one defining and one implementing declaration.
+- Is not permitted to have the `abstract` modifier.
+- Cannot explicitly implement an interface member.
+
+A partial event is not field-like ([§15.8.2][event-field-like]), i.e.:
+- It does not have any backing storage or accessors generated by the compiler.
+- It can only be used in `+=` and `-=` operations, not as a value.
+
+A defining partial constructor declaration:
+- Cannot have a constructor initializer (`: this()` or `: base()`; [§15.11.2][ctor-init]).
+- Does not perform any variable initializers ([§15.11.3][ctor-var-init]).
```

Comment:
I find it a little confusing to actually specify this, it makes me wonder if we give the same precaution about `extern` constructors.

Reply:
It is odd. If you wish, you could open an issue on csharpstandard.

Reply:
Opened https://github.com/dotnet/csharpstandard/issues/1253